frontend/src/components/module-editor/BlockPreview.tsx

```diff
@@ -13,14 +20,50 @@ const BlockStyle = ({
   className?: string;
   content: ContentTypeEnum;
 }): React.ReactElement => {
+  let contentIcon: React.ReactElement;
```

Comment:
Honestly, the switch statement is fine, but since you mentioned in your PR, one other option might be:

```javascript
const ICONS = {
    "column.svg": ColumnIcon,
    ...
};

// later
if (Object.keys(ICONS).includes(content.preview)) {
    const IconComponent = ICONS[content.preview];
    contentIcon = <IconComponent width="fit-content" height="fit-content" />;
} else {
    contentIcon = <ImageIcon />;
}
```

Reply:
Hm, will leave it as switch cases for now then. Maybe if we end up adding more content types we'll need to refactor accordingly